### ISSUE TITLE
Shell mode "edit" command raising error

### ISSUE THREAD

**Issue by ludocrous** [laurent22/joplin#117]
### Operating system

- [ ] Windows
- [x] macOS (10.13.2)
- [ ] Linux
- [ ] Android
- [ ] iOS

### Application

- [ ] Desktop
- [ ] Mobile
- [x] Terminal - joplin 0.10.83 (prod)

*If it is a bug if possible please provide a screenshot showing the problem*

*If there is an error, please copy and paste the full error message*

**app(...).gui(...).forceRender is not a function.**

---

From the terminal (bash):

`$ joplin ls -l`

will show the note list like:
`b7f33 20/12/2017 20:27 SSH`

Things like 

`$ joplin set b7f33 title "SSH Notes" `

OR 

`$ joplin ren "SSH" "SSH Notes" `

work fine, but the trying to edit the note using

`$ joplin edit b7f33`

OR 

`$ joplin edit "SSH" `

Both yield the following error: "app(...).gui(...).forceRender is not a function"

Editor config is set to: **"subl -w"** and the edit command works inside the terminal app itself, just not from the shell command line.

**Comment by laurent22** [comment]
Right, it's quite a simple fix, it's the dummyGui that needs a `forceRender()` method (that does nothing) in there: https://github.com/laurent22/joplin/blob/master/CliClient/app/app.js#L275

I will look at it once other things are out of the way, or someone can create a pull request.

**Comment by ludocrous** [comment]
Thanks.

Just FYI in case it helps,  I am not a JS developer at all but had a go at adding the missing method.
Issue then moves:

**app(...).gui(...).term is not a function**

Added a function called "term" as well, then got:

**Cannot read property 'saveState' of undefined**

So will leave it to you ....... :-) Great tool btw

**Comment by laurent22** [comment]
Yes a relatively quick fix too. Just writing the lines here for later reference:

```
CliClient\app\command-edit.js:81
CliClient\app\command-edit.js:86
```

**Comment by laurent22** [comment]
That should be fixed in the latest release on npm - version 0.10.84
